mobile/cmp/input/TextInput.js

```diff
@@ -96,31 +108,48 @@ const cmp = hoistCmp.factory(
     ({model, className, ...props}, ref) => {
         const {width, ...layoutProps} = getLayoutProps(props);
 
-        return input({
-            value: model.renderValue || '',
-
-            autoComplete: withDefault(props.autoComplete, props.type === 'password' ? 'new-password' : 'off'),
-            disabled: props.disabled,
-            modifier: props.modifier,
-            placeholder: props.placeholder,
-            spellCheck: withDefault(props.spellCheck, false),
-            tabIndex: props.tabIndex,
-            type: props.type,
-
-            style: {
-                ...props.style,
-                ...layoutProps,
-                width: withDefault(width, null),
-                textAlign: withDefault(props.textAlign, 'left')
-            },
-
-            onChange: model.onChange,
-            onKeyDown: model.onKeyDown,
-            onBlur: model.onBlur,
-            onFocus: model.onFocus,
-
+        return hbox({
+            ref,
             className,
-            ref
-        });
+            items: [
+                input({
+                    value: model.renderValue || '',
+
+                    autoComplete: withDefault(props.autoComplete, props.type === 'password' ? 'new-password' : 'off'),
+                    disabled: props.disabled,
+                    modifier: props.modifier,
+                    placeholder: props.placeholder,
+                    spellCheck: withDefault(props.spellCheck, false),
+                    tabIndex: props.tabIndex,
+                    type: props.type,
+
+                    style: {
+                        ...props.style,
+                        ...layoutProps,
+                        width: withDefault(width, null),
+                        textAlign: withDefault(props.textAlign, 'left')
+                    },
+
+                    onChange: model.onChange,
+                    onKeyDown: model.onKeyDown,
+                    onBlur: model.onBlur,
+                    onFocus: model.onFocus,
+                }),
+                model.showClearButton ? clearButton() : null
```

Comment:
total nit, but might use omit for readability here.

Reply:
Good point, change pushed.